.github/workflows/run-tests.yml

```diff
@@ -145,21 +163,49 @@ jobs:
         run: |
           docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} down -v
 
+
+  set-up-functional-test:
+    name: Set up functional test
+    runs-on: ubuntu-20.04
+    outputs:
+      shard_indices: ${{ steps.generate_output.outputs.shard_indices }}
+      total_shards: ${{ steps.generate_output.outputs.shards }}
+      github_timeout: ${{ steps.generate_output.outputs.github_timeout }}
+      test_timeout: ${{ steps.generate_output.outputs.test_timeout }}
+      single_test_args: ${{ steps.generate_output.outputs.single_test_args }}
+    steps:
+      - id: generate_output
+        run: |
+          shards=3
+          timeout=30
+          if [[ "${{ inputs.run_single_functional_test }}" == "true" ]]; then
+            shards=1
+            timeout=${{ inputs.timeout_minutes }}
+            single_test_args="-test.run ${{ inputs.test_name }} -count ${{ inputs.n_runs }}"
```

Comment:
why `-test.run` but `-count`? shouldn't they be both or neither `-test.` flags?

Reply:
honestly I just first learned to run testify tests with `go test ./tests -test.run TestFunctionalSuite -testify.m SubTestName` so that's what I used here..

looks like `go test ./tests -run TestFunctionalSuite/SubTestName` is equivalent, and that looks nicer, so I'll replace